Create an endpoint to add songs to an existing playlist

bbauman1 opened the pull request:
I couldn't find an existing function to hit this url

The response is an empty json and I couldn't think of any value in adding a return type to the function

rudrankriyam commented:
Hi @bbauman1! Thank you so much for adding this! TIL about this endpoint. The most we can do is check the response code with 201 and return a boolean value. So, we can use it like: 

```swift 
let result = try await MusadoraKit.addSongs(songs, toPlaylist: id) 

if result { 
// update U.I.
}
```

But it is already good as it is. Another thing I've realized which is convenient in autocompletion is passing the `Playlist` object directly? 

```swift 
static func add(songs: Songs, to playlist: Playlist) async throws {
}
```

Usage: 

```swift 
try await MusadoraKit.add(songs: songs, to: playlist) 
```

What do you think? 😄 And thank you so much for letting me know about this endpoint!!

> P.S. I'm working on [MusadoraKit 2.0](https://github.com/rryam/MusadoraKit/pull/16) with breaking changes, and I want to let you know beforehand.

bbauman1 commented:
@rudrankriyam Hey I went ahead and split the function into two different functions. 1 that uses the full `Song` and `Playlist` types and 1 that only requires IDs. 

So that you get an idea of my use case, my app allows multiple music services and I convert the MusicKit.Song to a generic "Song" type. When creating a playlist I'm not able to recreate a full MusicKit.Song object, I only have the ID.

Also added the bool return type and made it discardable
